services/website/website.go

```diff
@@ -157,16 +163,38 @@ func (srv *Webserver) updateStatusHTMLData() {
 	numKnown := printer.Sprintf("%d", len(knownValidators))
 	numPayloads := printer.Sprintf("%d", _numPayloadsDelivered)
 	latestSlot := printer.Sprintf("%d", _latestSlotInt)
+	relayHash := srv.getRelayHash()
 
 	srv.statusHTMLDataLock.Lock()
 	srv.statusHTMLData.ValidatorsTotal = numKnown
 	srv.statusHTMLData.ValidatorsRegistered = numRegistered
 	srv.statusHTMLData.Payloads = payloads
 	srv.statusHTMLData.HeadSlot = latestSlot
 	srv.statusHTMLData.NumPayloadsDelivered = numPayloads
+	srv.statusHTMLData.RelayHash = relayHash
 	srv.statusHTMLDataLock.Unlock()
 }
 
+func (srv *Webserver) getRelayHash() string {
+	if srv.cachedRelayHash != nil {
+		return *srv.cachedRelayHash
+	}
+	exe, err := os.Executable()
+	if err != nil {
+		srv.log.WithError(err).Error("error getting executable")
+		return ""
+	}
+	content, err := os.ReadFile(exe)
+	if err != nil {
+		srv.log.WithError(err).Error("error reading executable")
+		return ""
+	}
+	sum := sha256.Sum256(content)
+	srv.cachedRelayHash = new(string)
```

Comment:
is this a possible race condition with concurrent read & write?

Reply:
Yes, I think it is! Good spot.